includes/integrations/maxmind-geolocation/class-wc-integration-maxmind-geolocation-database.php

```diff
@@ -0,0 +1,115 @@
+<?php
+/**
+ * WC Integration MaxMind Geolocation Database
+ *
+ * @version 3.9.0
+ * @package WooCommerce/Integrations
+ */
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * WC Integration MaxMind Geolocation Database
+ *
+ * @version 3.9.0
+ */
+class WC_Integration_MaxMind_Geolocation_Database {
+
+	/**
+	 * The name of the MaxMind database to utilize.
+	 */
+	const DATABASE = 'GeoLite2-Country';
+
+	/**
+	 * The extension for the MaxMind database.
+	 */
+	const DATABASE_EXTENSION = '.mmdb';
+
+	/**
+	 * Fetches the path that the database should be stored.
+	 *
+	 * @return string The local database path.
+	 */
+	public static function get_database_path() {
+		$database_path = WP_CONTENT_DIR . '/uploads/' . self::DATABASE . self::DATABASE_EXTENSION;
+
+		/**
+		 * Filter the geolocation database storage path.
+		 *
+		 * @param string $database_path The path to the database.
+		 * @param int $version Deprecated since 3.4.0.
+		 * @deprecated 3.9.0
+		 */
+		$database_path = apply_filters_deprecated(
+			'woocommerce_geolocation_local_database_path',
+			array( $database_path, 2 ),
+			'3.9.0',
+			'woocommerce_maxmind_geolocation_database_path'
+		);
+
+		/**
+		 * Filter the geolocation database storage path.
+		 *
+		 * @since 3.9.0
+		 * @param string $database_path The path to the database.
+		 */
+		return apply_filters( 'woocommerce_maxmind_geolocation_database_path', $database_path );
+	}
+
+	/**
+	 * Fetches the database from the MaxMind service.
+	 *
+	 * @param string $license_key The license key to be used when downloading the database.
+	 * @return string|WP_Error The path to the database file or an error if invalid.
+	 */
+	public static function download_database( $license_key ) {
+		$download_uri = add_query_arg(
+			array(
+				'edition_id'  => self::DATABASE,
+				'license_key' => $license_key,
```

Comment:
I think we could sanitize this `license_key` with `wc_clean()`, or the escape the URL with `esc_url_raw()`.

Reply:
Ah, I didn’t realize that the query arguments wouldn’t be escaped this way. Arguably, I don’t actually think in this case that there are any security problems though, and the license keys are ASCII it appears. Can’t hurt to add though, of course.